rfc/plugins.md

```diff
@@ -0,0 +1,343 @@
+Whether plugins are implemented as in-process or not, Front50 should be used to track and expose a single source of truth of what plugins are installed in any particular service, as well as which ones are enabled or disabled.
+
+The Front50 plugin registry will also be exposed through Gate, and can be consumed by Deck to load the correct plugins on application startup.
+
+Plugins can also be enabled or disabled at runtime without redeploying a service. 
+This flag is sourced from Front50, which all services will be listening for change events from. Whether this change event is published or polled for initially is unknown.
+
+Halyard configs can be used to inform Spinnaker of desired state, but whatever state is defined in Front50 will be used as the canonical source for plugins. 
+When a service starts up, it will ask Front50 what plugins need to be installed and which of those needs to enabled or disabled. Halyard’s role will only go as far as updating Front50’s repository of plugin metadata.
+
+Using Front50 as a source of truth will require extension of PF4J.
+
+```proto
+syntax = "proto3";
+
+// PluginManifest is used during registration of a plugin
+message PluginManifest {
+	// The name of the plugin, which is used as a fully 
+	// qualified identifier, including a namespace and 
+	// plugin name, ex: "netflix/fast-properties"
+	string name = 1;
+
+	// A short description of the plugin
+	// ex: "Provides rollout safety of runtime application 
+	// config via pipelines"
+	string description = 2;
+
+	// The name of the plugin author, ex: "Netflix"
+	string provider_name = 3;
+
+	// The semver plugin version, ex: "1.0.0"
+	string version = 4;
+
+	// A list of plugin dependencies. Initially this will be
+	// informational only. ex: "netflix/somedep>=1.x"
+	repeated string dependencies = 5;
+
+	// required_versions is a repeated string of service version	// requirements. e.g. "clouddriver >= 5.33.0"
+	repeated string required_versions = 6;
+
+	// A list of extension points that the plugin implements.
+	// e.g. "pipelineStage"
+	repeated string provided_capabilities = 7;
+
+	// Whether or not the plugin is enabled. An unset value 
+	// should be considered as true.
+	boolean enabled = 8;
+
+	// If the plugin is in-process and this flag is set to true,
+	// the plugin will be embedded into the process without
+	// isolation, allowing full modification of the application
+	// lifecycle. This is inherently unsafe and caution must be
+	// used setting this field.
+	boolean inprocess_unsafe = 9;
+}
+```
+
+#### Sending Plugin Configuration to Deck
+While Front50 will be the source of truth for which plugins are installed and enabled, we can pass that data via Deck’s configuration file (`settings.js`), we can pass that data via `settings.js` today and transition to using Front50 when it is ready. 
```

Comment:
grammar: there is some repeatedness/confusionness happening here, i think you can scratch the first ~, we can pass that data via Deck’s configuration file (`settings.js`)~